### PR TITLE
Automate release notes for both app versions

### LINKED ISSUE #215: Automated creation of release notes for both app versions
Currently the release notes are only available for the `stable`  version. That leads to confusion for users of the `dev` version because with every `dev` release between `stable` releases, they get the same _Whats New_' dialog. Furthermore, the release notes are handwritten which may be OK for occasional releases but especially for the `dev` version, an automated flow would be better. 

**Proposal:**
* Different releases notes for `dev`  and `stable`. Where the notes for `dev` version are the commit messages between the releases and the releases notes for the `stable` version consist of the issues or PRs tagged to this version (milestone).
* The releases notes for the `dev` version is generated during the compile process of the app and will only be included in the app bundle. On the other hand, the release notes for the `stable` version are additionally committed to the repo for further usage e.g. on the website.
* The format for the notes will be _Markdown_ as already implemented.

**Requirements:**
* PRs and issues must have a reasonable subject, label and milestone.
* Commit message need a proper format as already stated in the contribution doc

### PR DESCRIPTION
**Pull request by chrgernoe**
Fixes #215

### CODE REVIEW

**Comment by chrgernoe** [comment]
I think the `Release Drafter` fails because the config file is not yet found in the master branch. I tested the GitHub Action on a playground repository and there it worked.

File: app/src/main/java/com/yacgroup/yacguide/utils/MarkdownUtils.kt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2020 Christian Sommer
+ * Copyright (C) 2020 2022 Christian Sommer
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/utils/MarkdownUtils.kt:2]: Is it best practice to collect all years, the file has changed? I would simply go for the most recent year.
Sth. like `2020, 2022, 2023, 2025` looks strange.

**Review by paetz — DISMISSED** [review]

File: app/src/main/java/com/yacgroup/yacguide/WhatsNewInfo.kt [hunk 2]
```diff
@@ -36,24 +35,28 @@ class WhatsNewInfo(private var _activity: AppCompatActivity) {
     }
 
     private var _sharedPrefs: SharedPreferences
-    private var _numReleaseNotesShow: Int = 1
     private var _releaseNoteFiles: Array<String>?
     private var _markwonBuilder: Markwon
     private lateinit var _view: View
+    private var _activityUtils: ActivityUtils
```
[the remark this answers is not in the document]
Comment (chrgernoe) [reply app/src/main/java/com/yacgroup/yacguide/WhatsNewInfo.kt:41]: I made also changed the other `private var` definitions to `private val`, except for the `_view`.

File: app/src/main/java/com/yacgroup/yacguide/utils/MarkdownUtils.kt (hunk 1 above)
Comment (chrgernoe) [reply app/src/main/java/com/yacgroup/yacguide/utils/MarkdownUtils.kt:2]: I'm not an expert but I think you need to list all years or a range: https://www.gnu.org/licenses/gpl-howto.en.html

File: docs/_release-notes/v0.5.0.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+## Important notice
+
+* Es müssen alle bereits heruntergeladenen Gebiete aktualisiert werden
+  (Gebiete verwalten -> Alle aktualisieren). Andernfalls ist
+  eine funktionierende Volltextsuche, sowie die korrekte Anzeige von
+  Felsen/Wegen nicht gewährleistet.
+
+## What's Changed
+
+* Add username to comment ([#254](https://github.com/yacgroup/yacguide/pull/254))
```
[the remark this answers is not in the document]
Comment (chrgernoe) [reply docs/_release-notes/v0.5.0.md:15]: I agree, we can rework the existing ones.

**Pull request merged**